tracing-appender/src/writer.rs

```diff
@@ -0,0 +1,115 @@
+use crate::rolling::create_writer_file;
+use std::fs::File;
+use std::io;
+
+#[cfg(feature = "compression_gzip")]
+use {
+    crate::compression::CompressionConfig,
+    flate2::write::GzEncoder,
+    std::io::BufWriter,
+    std::sync::{Arc, RwLock},
+};
+
+#[cfg(feature = "compression_gzip")]
+#[derive(Debug)]
+pub(crate) struct CompressedGzip {
+    compression: CompressionConfig,
+    buffer: Arc<RwLock<BufWriter<GzEncoder<BufWriter<File>>>>>,
+}
+
+#[derive(Debug)]
+pub(crate) enum WriterChannel {
+    File(File),
+    #[cfg(feature = "compression_gzip")]
+    CompressedFileGzip(CompressedGzip),
+}
```

Comment:
nit, take it or leave it: we probably don't _really_ need a separate struct for the GZIP variant of `WriterChannel`, since it has no methods of its own...this could probably be
```suggestion
#[cfg(feature = "compression_gzip")]
#[derive(Debug)]
pub(crate) struct CompressedGzip 

#[derive(Debug)]
pub(crate) enum WriterChannel {
    File(File),
    #[cfg(feature = "compression_gzip")]
    CompressedFileGzip {
        compression: CompressionConfig,
        buffer: Arc<RwLock<BufWriter<GzEncoder<BufWriter<File>>>>>,
    }
}
```

but, maybe having it in its own struct is a little bit neater-looking? i could be convinced either way.